src/components/contextual/stake/StakePreview.vue

```diff
@@ -81,7 +86,7 @@ const isActionConfirmed = ref(false);
 const confirmationReceipt = ref<TransactionReceipt>();
 const stakeActions = ref<TransactionActionInfo[]>([]);
 const shareBalanceToDisplay = ref(
-  props.action === 'unstake'
+  props.action === 'unstake' || props.action === 'restake'
     ? stakedSharesForProvidedPool.value
     : balanceFor(props.pool.address)
 );
```

Comment:
I understand that this was already set as a `ref` but this also looks like it should be a computed property. My understanding is that `ref`s are for properties that you need to be mutable, i.e. be able to modify elsewhere in the code. I'm pretty sure `shareBalanceToDisplay` does not need to be modified but it does need to be reactive based on the internal properties it depends on. Therefore it should be a computed property.